other/sdes.py

```diff
@@ -0,0 +1,86 @@
+def apply_table(inp, table):
+    res = ''
+    for i in table:
+        res += inp[i-1]
+    return res
+
+def left_shift(data):
+    li = []
+    for i in range(len(data)-1):
+        li.append(data[i+1])
+    li.append(data[0])
+    return ''.join(li)
+
+def XOR(a, b):
+    res = ''
+    for i in range(len(a)):
+        if a[i]==b[i]:
+            res += '0'
+        else:
+            res += '1'
+    return res
+
+def apply_sbox(s, data):
+    row = int('0b'+data[0]+data[-1], 2)
+    col = int('0b'+data[1:3], 2)
+    return bin(s[row][col])[2:]
+
+def function(expansion, s0, s1, key, message):
+    left = message[:4]
+    right = message[4:]    
+    temp = apply_table(right, expansion)
+    temp = XOR(temp, key)
+    l = apply_sbox(s0, temp[:4])
+    r = apply_sbox(s1, temp[4:])
+    l = '0'*(2-len(l)) + l
+    r = '0'*(2-len(r)) + r
+    temp = apply_table(l+r, p4_table)
+    temp = XOR(left, temp)
+    return temp+right
+    
+

```

Comment:
Insert: `if __name__ == "__main__":
And then indent all remaining lines under that to enable our automated tests to pass as discussed in CONTRIBUTING.md